internal/dependency/manager/manager_test.go

```diff
@@ -0,0 +1,288 @@
+package manager
+
+import (
+	"github.com/g2a-com/klio/internal/context"
+	"github.com/g2a-com/klio/internal/dependency"
+	"github.com/spf13/afero"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+)
+
+//==========================================================================================
+//::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+//==========================================================================================
+// Take a look at DEP INDEX in utils_test.go
+//==========================================================================================
+
+func TestGetInstalledCommands(t *testing.T) {
+	tests := []struct {
+		// Name of the test
+		name string
+		// Injected mock function for returning local dependencies
+		fetchIndexFunction func(filePath string) (*dependency.DependenciesIndex, error)
+		// Configuration paths for klio
+		paths context.Paths
+		// Desired index list of dependencies installed
+		want []dependency.DependenciesIndexEntry
+	}{
+		{
+			name:               "errorFromIndex",
+			fetchIndexFunction: fetchInvalidIndex,
+			paths: context.Paths{
+				ProjectInstallDir: validProjectInstallPath,
+				GlobalInstallDir:  validGlobalInstallPath,
+			},
+			want: nil,
+		},
+		{
+			name:               "emptyDirConfig",
+			fetchIndexFunction: fetchInvalidIndex,
+			paths: context.Paths{
+				ProjectInstallDir: "",
+				GlobalInstallDir:  "",
+			},
+			want: nil,
+		},
+		{
+			name:               "simpleValidIndex",
+			fetchIndexFunction: fetchSimpleValidIndex,
+			paths: context.Paths{
+				ProjectInstallDir: validProjectInstallPath,
+				GlobalInstallDir:  validGlobalInstallPath,
+			},
+			want: properlyProcessedValidSimpleIndex,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Please structure tests using [Arrange-Act-Assert](https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/) pattern.

Reply:
This buzz-phrase means nothing to me. The article shows the pattern that most of the tests follow. You need them to be split into separate functions? There is a comment on that already. Please explain.